--- scl-utils-2.0.2/rpm/macros.scl.old	2019-07-04 09:27:29.887470341 +0200
+++ scl-utils-2.0.2/rpm/macros.scl	2019-07-30 10:24:32.536648421 +0200
@@ -93,7 +93,7 @@
     /usr/lib/rpm/brp-strip-comment-note %{__strip} %{__objdump}
     }
     /usr/lib/rpm/brp-strip-static-archive %{__strip}
-    /usr/lib/rpm/brp-scl-python-bytecompile %{__python} %{?_python_bytecompile_errors_terminate_build} %{_scl_root}
+    /usr/lib/rpm/brp-scl-python-bytecompile %{__python3} %{?_python_bytecompile_errors_terminate_build} %{_scl_root}
     /usr/lib/rpm/brp-python-hardlink
 %{nil}}
 BuildRequires: scl-utils-build
