diff --git a/ext/session/session4.test b/ext/session/session4.test
index de183a6..0b4af9e 100644
--- a/ext/session/session4.test
+++ b/ext/session/session4.test
@@ -11,8 +11,6 @@
 # This file implements regression tests for the session module. 
 # 
 
-package require Tcl 8.6
-
 if {![info exists testdir]} {
   set testdir [file join [file dirname [info script]] .. .. test]
 } 
diff --git a/test/zipfile.test b/test/zipfile.test
index 2bab066..42e7932 100644
--- a/test/zipfile.test
+++ b/test/zipfile.test
@@ -10,8 +10,6 @@
 #***********************************************************************
 #
 
-package require Tcl 8.6
-
 set testdir [file dirname $argv0]
 source $testdir/tester.tcl
 set testprefix zipfile
diff --git a/test/zipfile2.test b/test/zipfile2.test
index 987e4f0..897d92b 100644
--- a/test/zipfile2.test
+++ b/test/zipfile2.test
@@ -10,8 +10,6 @@
 #***********************************************************************
 #
 
-package require Tcl 8.6
-
 set testdir [file dirname $argv0]
 source $testdir/tester.tcl
 set testprefix zipfile2
