From: Christian Hofstaedtler <zeha@debian.org>
Date: Thu, 15 Mar 2018 14:51:43 +0100
Subject: Replace bundled libraries with system versions
Forwarded: not-needed

--- a/lib/bundler/vendored_molinillo.rb
+++ b/lib/bundler/vendored_molinillo.rb
@@ -1,4 +1,4 @@
 # frozen_string_literal: true
 
 module Bundler; end
-require_relative "vendor/molinillo/lib/molinillo"
+require "molinillo"
--- a/lib/bundler/vendored_persistent.rb
+++ b/lib/bundler/vendored_persistent.rb
@@ -15,10 +15,10 @@
     end
   end
 end
-require_relative "vendor/net-http-persistent/lib/net/http/persistent"
+require "net/http/persistent"
 
 module Bundler
-  class PersistentHTTP < Persistent::Net::HTTP::Persistent
+  class PersistentHTTP < Net::HTTP::Persistent
     def connection_for(uri)
       super(uri) do |connection|
         result = yield connection
--- a/lib/bundler/vendored_thor.rb
+++ b/lib/bundler/vendored_thor.rb
@@ -2,7 +2,7 @@
 
 module Bundler
   def self.require_thor_actions
-    require_relative "vendor/thor/lib/thor/actions"
+    require "thor/actions"
   end
 end
-require_relative "vendor/thor/lib/thor"
+require "thor"
